# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xinit/startwm.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

More flexible root application startup.

--- xinit-0.99.4/xinitrc.cpp_orig	2005-11-15 05:03:10.000000000 +0100
+++ xinit-0.99.4/xinitrc.cpp	2005-12-15 01:42:29.000000000 +0100
@@ -24,7 +24,7 @@
     XMODMAP $usermodmap
 fi
 
-XCOMM start some nice programs
+XCOMM start windowmanager or some programs 
 
 #if defined(__SCO__) || defined(__UNIXWARE__)
 if [ -r /etc/default/xdesktops ]; then
@@ -68,8 +68,9 @@
 
 XCOMM This is the fallback case if nothing else is executed above
 #endif /* !defined(__SCO__)  && !defined(__UNIXWARE__) */
-TWM &
-XCLOCK -geometry 50x50-1+1 &
-XTERM -geometry 80x50+494+51 &
-XTERM -geometry 80x20+494-0 &
-exec XTERM -geometry 80x66+0+0 -name login
+
+WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+xsetroot -solid darkcyan ; exec twm ; exec xterm
